swap_meet/vendor.py

```diff
@@ -1,2 +1,115 @@
+from .item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self,inventory = None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+        
+
+    
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+           
+
+    def get_by_category(self,category):
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+            
+
+    def swap_items(self,other,my_item,their_item):
+        if my_item in self.inventory and their_item in other.inventory:
+            self.inventory.remove(my_item)
+            other.inventory.append(my_item)
+            other.inventory.remove(their_item) 
+            self.inventory.append(their_item)
```

Comment:
I recommend using the vendor methods `add` and `remove` here.
```suggestion
            self.remove(my_item)
            other.add(my_item)
            other.remove(their_item) 
            self.add(their_item)
```